### PR TITLE
Improve WS logging

### PR DESCRIPTION
**Pull request by boxhock**

### CODE REVIEW

**Comment by boxhock** [comment]
Example **INFO** level logging:

```
{"message":"Listening on port 8080!","level":"info","instanceId":"60c8c74f-dbef-4b8c-9993-a9842f502e5d","timestamp":"2021-05-07T10:46:58.561Z"}
{"url":"wss://ws-feed.pro.coinbase.com","key":"1","level":"info","message":"WS: New connection","instanceId":"60c8c74f-dbef-4b8c-9993-a9842f502e5d","timestamp":"2021-05-07T10:47:01.698Z"}
{"connection_key":"1","connection_url":"wss://ws-feed.pro.coinbase.com","feed_id":"ETH/USD","subscription_key":"29dee2a5b28540a32b8b35e484187f00cff15043","level":"info","message":"WS: Subscribed","instanceId":"60c8c74f-dbef-4b8c-9993-a9842f502e5d","timestamp":"2021-05-07T10:47:01.807Z"}
{"connection_key":"1","connection_url":"wss://ws-feed.pro.coinbase.com","feed_id":"ETH/USD","subscription_key":"29dee2a5b28540a32b8b35e484187f00cff15043","level":"info","message":"WS: Unsubscribed","instanceId":"60c8c74f-dbef-4b8c-9993-a9842f502e5d","timestamp":"2021-05-07T10:48:11.227Z"}
{"url":"wss://ws-feed.pro.coinbase.com","key":"1","level":"info","message":"WS: Disconnected","instanceId":"60c8c74f-dbef-4b8c-9993-a9842f502e5d","timestamp":"2021-05-07T10:48:11.334Z"}
```

**Comment by boxhock** [comment]
Example **DEBUG** level logging:

```
{"message":"Listening on port 8080!","level":"info","instanceId":"097912dc-e747-431f-afe0-57044aab5599","timestamp":"2021-05-07T10:50:49.118Z"}
{"input":{"id":"test","data":{"from":"ETH","to":"USD"},"debug":{"feedId":"ETH/USD"}},"level":"debug","message":"Input: ","instanceId":"097912dc-e747-431f-afe0-57044aab5599","timestamp":"2021-05-07T10:50:50.244Z"}
{"config":{"api":{"withCredentials":false,"baseURL":"https://api.coinbase.com","timeout":30000},"verbose":false},"level":"debug","message":"Adapter configuration:","instanceId":"097912dc-e747-431f-afe0-57044aab5599","timestamp":"2021-05-07T10:50:50.244Z"}
{"message":"Received response","data":{"data":{"base":"ETH","currency":"USD","amount":"3460.02"}},"status":200,"statusText":"OK","level":"debug","instanceId":"097912dc-e747-431f-afe0-57044aab5599","timestamp":"2021-05-07T10:50:50.459Z"}
{"output":{"result":3460.02},"level":"debug","message":"Output: [200]: ","instanceId":"097912dc-e747-431f-afe0-57044aab5599","timestamp":"2021-05-07T10:50:50.459Z"}
{"url":"wss://ws-feed.pro.coinbase.com","key":"1","level":"info","message":"WS: New connection","instanceId":"097912dc-e747-431f-afe0-57044aab5599","timestamp":"2021-05-07T10:50:50.683Z"}
{"connection_key":"1","connection_url":"wss://ws-feed.pro.coinbase.com","feed_id":"ETH/USD","subscription_key":"29dee2a5b28540a32b8b35e484187f00cff15043","level":"info","message":"WS: Subscribed","instanceId":"097912dc-e747-431f-afe0-57044aab5599","timestamp":"2021-05-07T10:50:50.794Z"}
[...]
{"feed_id":"ETH/USD","subscription_key":"29dee2a5b28540a32b8b35e484187f00cff15043","level":"debug","message":"WS: Got message","instanceId":"097912dc-e747-431f-afe0-57044aab5599","timestamp":"2021-05-07T10:52:00.149Z"}
{"payload":{"type":"subscribe","channels":["ticker"],"product_ids":["ETH-USD"]},"level":"debug","message":"WS: unsubscribe (inactive feed)","instanceId":"097912dc-e747-431f-afe0-57044aab5599","timestamp":"2021-05-07T10:52:00.243Z"}
{"connection_key":"1","connection_url":"wss://ws-feed.pro.coinbase.com","feed_id":"ETH/USD","subscription_key":"29dee2a5b28540a32b8b35e484187f00cff15043","level":"info","message":"WS: Unsubscribed","instanceId":"097912dc-e747-431f-afe0-57044aab5599","timestamp":"2021-05-07T10:52:00.245Z"}
{"url":"wss://ws-feed.pro.coinbase.com","key":"1","level":"info","message":"WS: Disconnected","instanceId":"097912dc-e747-431f-afe0-57044aab5599","timestamp":"2021-05-07T10:52:00.355Z"}
```

**Comment by boxhock** [comment]
Example **DEBUG=true** logging:

```
{"message":"Listening on port 8080!","level":"info","instanceId":"68534bdf-b76c-47b2-9232-20c24aa308c2","timestamp":"2021-05-07T10:54:05.063Z"}
 action WS/CONNECT! @ 12:54:08.017
   prev state {
    cacheWarmer: { subscriptions: {}, warmups: {} },
    rateLimit: { heartbeats: { total: [Object], participants: {} } },
    ws: {
      connections: { active: {}, connecting: {}, total: 0 },
      subscriptions: {}
    }
  }
   action     {
    type: 'WS/CONNECT!',
    payload: {
      id: 'MXenIxUDPf3plm90SDLC1',
      createdAt: '2021-05-07T10:54:08.017Z',
      config: {
        connectionInfo: [Object],
        connectionLimit: 1,
        connectionTTL: 70000,
        connectionRetryLimit: 3,
        connectionRetryDelay: 1000,
        subscriptionLimit: 10,
        subscriptionTTL: 70000,
        subscriptionUnresponsiveTTL: 70000,
        subscriptionPriorityList: []
      },
      wsHandler: {
        connection: [Object],
        subscribe: [Function: subscribe],
        unsubscribe: [Function: unsubscribe],
        subsFromMessage: [Function: subsFromMessage],
        isError: [Function: isError],
        filter: [Function: filter],
        toResponse: [Function: toResponse]
      }
    }
  }
   next state {
    cacheWarmer: { subscriptions: {}, warmups: {} },
    rateLimit: { heartbeats: { total: [Object], participants: {} } },
    ws: {
      connections: { active: {}, connecting: [Object], total: 0 },
      subscriptions: {}
    }
  }
[...]
```

**Review by HenryNguyen5 — CHANGES_REQUESTED** [review]

File: packages/core/bootstrap/src/lib/store/index.ts [hunk 1]
```diff
@@ -34,7 +34,7 @@ export function configureStore(
   preloadedState: PreloadedState<any> = {},
   middleware: Middleware<unknown, any, Dispatch<AnyAction>>[] = [],
 ): Store {
-  if (isDebug() || isDebugLogLevel()) middleware.push(logger)
+  if (isDebug()) middleware.push(logger)
```
[the remark this answers is not in the document]
Comment (boxhock) [reply packages/core/bootstrap/src/lib/store/index.ts:37]: Right, I think there's simply too much information being logged to make sense of it when you can connect a debugger to it instead. Should we remove it in this PR or track it in a separate story?

[the remark this answers is not in the document]
Comment (HenryNguyen5) [reply packages/core/bootstrap/src/lib/store/index.ts:37]: just remove it, it improves WS logging :D

File: packages/core/bootstrap/src/lib/ws/epics.ts [hunk 2]
```diff
@@ -355,6 +349,62 @@ export const metricsEpic: Epic<AnyAction, AnyAction, any, any> = (action$, state
     filter(() => false), // do not duplicate events
   )
 
-export const rootEpic = combineEpics(connectEpic, metricsEpic)
+export const loggingEpic: Epic<AnyAction, AnyAction, any, any> = (action$, state$) =>
```
[the remark this answers is not in the document]
Comment (boxhock) [reply packages/core/bootstrap/src/lib/ws/epics.ts:352]: @HenryNguyen5 what's your opinion on moving some of the log lines inside the metrics epic? So for simple connect/disconnect, subscribe/unsubscribe we utilize the same filtering and all that, and for more advanced logging, like logging for how messages are handled, we do it inside the connect epic?

Keeping everything inside the connect epic seems to get too messy.

[the remark this answers is not in the document]
Comment (HenryNguyen5) [reply packages/core/bootstrap/src/lib/ws/epics.ts:352]: I really think we should keep it inside connect, by putting it inside metrics we're logging side-effects via side-effects, it feels like too much indirection to me. 

If it does get too messy, we can always make our own rx operators or helper functions that help us out in terms of logging noise

**Review by HenryNguyen5 — CHANGES_REQUESTED** [review]

**Review by justinkaseman — APPROVED** [review]
:1st_place_medal:

**Pull request merged**